presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQuery.java

```diff
@@ -94,7 +98,13 @@ public LocalDispatchQuery(
     public void startWaitingForResources()
     {
         if (stateMachine.transitionToWaitingForResources()) {
-            waitForMinimumWorkers();
+            if(isResourceManagerEnabled)
```

Comment:
I think the pattern here should be more along the line of chaining futures.  When we wait for the resource manager, we should also ensure the minimum coordinator count is met as well.

Reply:
the waitForMinimumResourceManager method does the same, once it finds enough resource manager, it triggers waitForMimimumWorkers method and wait for that.